### PR TITLE
truncate commit in version

### PR DESCRIPTION
**Pull request by BenTheElder**
xref discussion in https://github.com/kubernetes-sigs/kind/pull/988#discussion_r337336824

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -17,6 +17,7 @@ golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2 h1:VklqNMn3ovrHsnt90Pveol
 golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2/go.mod h1:djNgcEr1/C05ACkg1iLfiJU5Ep61QUkGW8qpdssI0+w=
 golang.org/x/crypto v0.0.0-20190611184440-5c40567a22f8 h1:1wopBVtVdWnn03fZelqdXTqk7U7zPQCb+T4rbU9ZEoU=
 golang.org/x/crypto v0.0.0-20190611184440-5c40567a22f8/go.mod h1:yigFU9vqHzYiE8UmvKecakEJjdnWj3jj499lnFckfCI=
+golang.org/x/net v0.0.0-20190404232315-eb5bcb51f2a3 h1:0GoQqolDA55aaLxZyTzK/Y2ePZzZTUrRacwib7cNsYQ=
```
Comment (BenTheElder) [inline go.sum]: not sure why this changed? will see if CI verify checks complain

File: cmd/kind/version/version.go [hunk 2]
```diff
@@ -35,7 +35,8 @@ func Version() string {
 		// if commit was set, add the + <build>
 		// we only do this for pre-release versions
 		if GitCommit != "" {
-			v += "+" + GitCommit
+			// NOTE: use 14 character short hash, like Kubernetes
+			v += "+" + truncate(GitCommit, 14)
```
Comment (BenTheElder) [inline cmd/kind/version/version.go]: Git says 12 characters is suficcient for the linux kernel, but kubernetes uses 14 and this still seems pretty short
https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection
https://github.com/kubernetes/kubernetes/blob/6bcfd75dd662625b2f92c41cc19fb0fd1cc5aa32/hack/lib/version.sh#L69

```
$ kind version
kind v0.6.0-alpha+cb1e6727f4d507 go1.13.3 linux/amd64
```
/shrug
we can always revisit and truncate more or less, this is trivial to change now.

File: go.sum (hunk 1 above)
Comment (neolit123) [reply go.sum]: LGTM, but not sure about this change too.

Comment (BenTheElder) [reply go.sum]: CI seems fine with it, and it's just one hash so /shrug
I'll follow up with taking another look at ensuring verify tidies

Comment (neolit123) [reply go.sum]: /lgtm

Comment (BenTheElder) [reply go.sum]: https://github.com/kubernetes-sigs/kind/issues/996

**Pull request merged**